langkit/lkt_lowering.py

```diff
@@ -1559,6 +1565,59 @@ def lower(expr: L.Expr) -> AbstractExpression:
                     }[type(expr.f_op)]
                     return E.Arithmetic(left, right, operator)
 
+            elif isinstance(expr, L.BlockExpr):
+                assert local_vars is not None
+
+                # Lower declarations for this block
+                vars = []
+                var_exprs = []
+
+                for v in expr.f_val_defs:
+                    if isinstance(v, L.ValDecl):
+                        source_name = names.Name.from_lower(v.f_syn_name.text)
+
+                        # Make sure the name used for code generation is a
+                        # valid Ada identifier.
+                        code_name = source_name.lower
```

Comment:
Having this here kind of makes me uncomfortable, shouldn't this be factorized in an independent abstraction ?

Reply:
Good idea, done!